pennylane_lightning/src/gates/DynamicDispatcher.hpp

```diff
@@ -137,6 +121,89 @@ template <typename PrecisionT> class DynamicDispatcher {
         return singleton;
```

Comment:
Assuming we are using multiple different instantiations across multiple threads, will this be fine? Or do we need to add any locking behaviour to anything in the given singleton?

Reply:
Always the same instance is used. Currently, we add all registrations when the program is loaded (in a single thread) and only use them afterward. As all other member functions besides registration functions in this class are thread-safe (they are const), we don't need mutex here unless there is any further design changes. I will add this comment to the description of this class.